cpu/stm32_common/periph/uart.c

```diff
@@ -294,7 +337,7 @@ static inline void irq_handler(uart_t uart)
     uint32_t status = dev(uart)->ISR;
 
     if (status & USART_ISR_RXNE) {
-        isr_ctx[uart].rx_cb(isr_ctx[uart].arg, (uint8_t)dev(uart)->RDR);
+        isr_ctx[uart].rx_cb(isr_ctx[uart].arg, (uint8_t)dev(uart)->RDR & data_mask);
```

Comment:
Does the 8th bit remain in the buffer if the config says only 7?

Reply:
Yes. Parity bit is the MSB and have to be masked to get the raw data.

Reply:
Can you read the register to see if it needs to be masked?  I think that is better than a static global (silly hardware though).

Reply:
I can make this.

Reply:
Nice!